File: app/src/main/java/com/kota/Bahamut/Dialogs/Dialog_SelectArticle.java
package com.kota.Bahamut.Dialogs;

import static com.kota.Bahamut.Service.CommonFunctions.getContextString;

import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.kota.ASFramework.Dialog.ASAlertDialog;
import com.kota.ASFramework.Dialog.ASDialog;
import com.kota.ASFramework.UI.ASToast;
import com.kota.Bahamut.Pages.BoardPage.BoardMainPage;
import com.kota.Bahamut.R;

import java.util.Objects;

public class Dialog_SelectArticle extends ASDialog implements View.OnClickListener {
    Button _button_0_p;
    Button _button_1_p;
    Button _button_2_p;
    Button _button_3_p;
    Button _button_4_p;
    Button _button_5_p;
    Button _button_6_p;
    Button _button_7_p;
    Button _button_8_p;
    Button _button_9_p;
    Button _button_back_space_p;
    Button _cancel_button;
    TextView _content;
    String _content_string = "";
    Dialog_SelectArticle_Listener _listener;
    Button _search_button;

    public String getName() {
        return "BahamutBoardSelectDialog";
    }

    public Dialog_SelectArticle() {
        requestWindowFeature(1);
        setContentView(R.layout.dialog_select_article);
        Objects.requireNonNull(getWindow()).setBackgroundDrawable(null);
        this._content = findViewById(R.id.Bahamut_Dialog_Select_content_Label);
        this._button_0_p = findViewById(R.id.Bahamut_Dialog_Select_Button_0_p);
        this._button_1_p = findViewById(R.id.Bahamut_Dialog_Select_Button_1_p);
        this._button_2_p = findViewById(R.id.Bahamut_Dialog_Select_Button_2_p);
        this._button_3_p = findViewById(R.id.Bahamut_Dialog_Select_Button_3_p);
        this._button_4_p = findViewById(R.id.Bahamut_Dialog_Select_Button_4_p);
        this._button_5_p = findViewById(R.id.Bahamut_Dialog_Select_Button_5_p);
        this._button_6_p = findViewById(R.id.Bahamut_Dialog_Select_Button_6_p);
        this._button_7_p = findViewById(R.id.Bahamut_Dialog_Select_Button_7_p);
        this._button_8_p = findViewById(R.id.Bahamut_Dialog_Select_Button_8_p);
        this._button_9_p = findViewById(R.id.Bahamut_Dialog_Select_Button_9_p);
        this._button_back_space_p = findViewById(R.id.Bahamut_Dialog_Select_Button_backSpace_p);
        this._search_button = findViewById(R.id.Bahamut_Dialog_Select_Button_Search);
        this._cancel_button = findViewById(R.id.Bahamut_Dialog_Select_Button_Cancel);
        this._button_0_p.setOnClickListener(this);
        this._button_1_p.setOnClickListener(this);
        this._button_2_p.setOnClickListener(this);
        this._button_3_p.setOnClickListener(this);
        this._button_4_p.setOnClickListener(this);
        this._button_5_p.setOnClickListener(this);
        this._button_6_p.setOnClickListener(this);
        this._button_7_p.setOnClickListener(this);
        this._button_8_p.setOnClickListener(this);
        this._button_9_p.setOnClickListener(this);
        this._button_back_space_p.setOnClickListener(this);
        this._search_button.setOnClickListener(this);
        this._cancel_button.setOnClickListener(this);
    }

    public void onClick(View view) {
        if (view == this._button_0_p) {
            this._content_string += 0;
        } else if (view == this._button_1_p) {
            this._content_string += 1;
        } else if (view == this._button_2_p) {
            this._content_string += 2;
        } else if (view == this._button_3_p) {
            this._content_string += 3;
        } else if (view == this._button_4_p) {
            this._content_string += 4;
        } else if (view == this._button_5_p) {
            this._content_string += 5;
        } else if (view == this._button_6_p) {
            this._content_string += 6;
        } else if (view == this._button_7_p) {
            this._content_string += 7;
        } else if (view == this._button_8_p) {
            this._content_string += 8;
        } else if (view == this._button_9_p) {
            this._content_string += 9;
        } else if (view == this._button_back_space_p) {
            if (this._content_string.length() > 0) {
                this._content_string = this._content_string.substring(0, this._content_string.length() - 1);
            }
        } else if (view == this._search_button) {
            if (this._content_string.isEmpty()) {
                ASToast.showShortToast(getContextString(R.string.please_input_article_number));
                return;
            }
            if (this._listener != null) {
                this._listener.onSelectDialogDismissWIthIndex(this._content_string);
            }
            dismiss();
        } else if (view == this._cancel_button) {
            dismiss();
        }
        if (this._content_string.length() > 5) {
            this._content_string = this._content_string.substring(0, 5);
        }
        this._content.setText(this._content_string);
    }

    public void setListener(Dialog_SelectArticle_Listener listener) {
        this._listener = listener;
    }

    public void show() {
        super.show();
    }
}

File: app/src/main/java/com/kota/Bahamut/dialogs/DialogSelectArticle.kt
package com.kota.Bahamut.dialogs

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.kota.Bahamut.R
import com.kota.Bahamut.service.CommonFunctions
import com.kota.Bahamut.ui.components.BahaText
import com.kota.Bahamut.ui.components.ButtonType
import com.kota.Bahamut.ui.dialogs.BahaAlertDialogContent
import com.kota.Bahamut.ui.dialogs.BahaDialogButton
import com.kota.Bahamut.ui.theme.AppTheme
import com.kota.asFramework.dialog.ASDialog
import com.kota.asFramework.ui.ASToast

class DialogSelectArticle : ASDialog() {
    var dialogSelectArticleListener: DialogSelectArticleListener? = null

    override val name: String?
        get() = "BahamutBoardSelectDialog"

    init {
        setTitle(CommonFunctions.getContextString(R.string.select_article))
        setComposeContent {
            Content()
        }
    }

    @Composable
    private fun Content() {
        var contentString by remember { mutableStateOf("") }
        val colors = AppTheme.colors

        fun onDigit(d: String) {
            if (contentString.length < 5) {
                contentString += d
            }
        }

        fun onBackspace() {
            if (contentString.isNotEmpty()) {
                contentString = contentString.dropLast(1)
            }
        }

        BahaAlertDialogContent(
            title = CommonFunctions.getContextString(R.string.select_article),
            contentPadding = PaddingValues(0.dp),
            buttons = listOf(
                BahaDialogButton(
                    text = CommonFunctions.getContextString(R.string.cancel),
                    type = ButtonType.DANGER,
                    onClick = { dismiss() }
                ),
                BahaDialogButton(
                    text = CommonFunctions.getContextString(R.string.search),
                    type = ButtonType.DANGER,
                    onClick = {
                        if (contentString.isEmpty()) {
                            ASToast.showShortToast(CommonFunctions.getContextString(R.string.please_input_article_number))
                            return@BahaDialogButton
                        }
                        dialogSelectArticleListener?.onSelectDialogDismissWIthIndex(contentString)
                        dismiss()
                    }
                )
            )
        ) {
            // 顯示目前輸入的數字
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(56.dp)
                    .padding(horizontal = 16.dp),
                contentAlignment = Alignment.Center
            ) {
                BahaText(
                    text = if (contentString.isEmpty()) CommonFunctions.getContextString(R.string.please_input_article_number) else contentString,
                    color = if (contentString.isEmpty()) colors.textSecondary else colors.textPrimary
                )
            }

            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(1.dp)
                    .background(colors.divider)
            )

            // 數字鍵盤
            val rows = listOf(
                listOf("7", "8", "9"),
                listOf("4", "5", "6"),
                listOf("1", "2", "3")
            )

            rows.forEach { row ->
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(48.dp)
                ) {
                    row.forEachIndexed { index, digit ->
                        if (index > 0) {
                            Box(
                                modifier = Modifier
                                    .width(1.dp)
                                    .fillMaxHeight()
                                    .background(colors.divider)
                            )
                        }
                        Box(
                            modifier = Modifier
                                .weight(1f)
                                .fillMaxHeight()
                                .background(colors.dialogBlockBackground)
                                .clickable { onDigit(digit) },
                            contentAlignment = Alignment.Center
                        ) {
                            BahaText(text = digit, color = colors.textPrimary)
                        }
                    }
                }
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(1.dp)
                        .background(colors.divider)
                )
            }

            // 0 與 DEL
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(48.dp)
            ) {
                Box(
                    modifier = Modifier
                        .weight(2f)
                        .fillMaxHeight()
                        .background(colors.dialogBlockBackground)
                        .clickable { onDigit("0") },
                    contentAlignment = Alignment.Center
                ) {
                    BahaText(text = "0", color = colors.textPrimary)
                }
                Box(
                    modifier = Modifier
                        .width(1.dp)
                        .fillMaxHeight()
                        .background(colors.divider)
                )
                Box(
                    modifier = Modifier
                        .weight(1f)
                        .fillMaxHeight()
                        .background(colors.dialogBlockBackground)
                        .clickable { onBackspace() },
                    contentAlignment = Alignment.Center
                ) {
                    BahaText(
                        text = CommonFunctions.getContextString(R.string.del),
                        color = colors.textPrimary
                    )
                }
            }
        }
    }

    fun setListener(listener: DialogSelectArticleListener?) {
        this.dialogSelectArticleListener = listener
    }
}
